crates/op-rbuilder/src/builders/flashblocks/payload.rs

```diff
@@ -214,6 +219,12 @@ where
 
         let chain_spec = self.client.chain_spec();
         let timestamp = config.attributes.timestamp();
+        let is_doing_historical_sync = timestamp < current_timestamp;
+        let calculate_state_root = if is_doing_historical_sync {
```

Comment:
why do we need to build blocks and calculate the state root during historical sync?